airflow/models/xcom.py

```diff
@@ -785,19 +801,26 @@ def __getstate__(self) -> Any:
             return (str(statement), query.count())
 
     def __setstate__(self, state: Any) -> None:
+        """Construct a query by retrieving the statement from the `state` and assign it to the `self._query`class property."""
         statement, self._len = state
         self._query = Query(XCom.value).from_statement(text(statement))
 
     def __len__(self):
+        """Declare a `self._len` class property that assigns the number of rows the query returns using `query.count()` if the `self._len` property is `None`.
+        
+        Otherwise return `self._len`
```

Comment:
Some more specific better here, `__len__` just about length of the object. Better avoid to detail explanation into the docstring. Implementation could be changed, and docstring transformed into the "liar"

You could ask a round in `#new-contributors` or `#contributors` slack channel (join link if you not there https://s.apache.org/airflow-slack)  if you would like what kind of the object it calculate length